src/main/java/io/openshift/booster/catalog/BoosterCatalogService.java

```diff
@@ -69,52 +70,58 @@
 
    private static final Logger logger = Logger.getLogger(BoosterCatalogService.class.getName());
 
-   private final ReentrantReadWriteLock reentrantLock = new ReentrantReadWriteLock();
-
-   private volatile List<Booster> boosters = Collections.emptyList();
+   private volatile Set<Booster> boosters = new ConcurrentSkipListSet<>(Comparator.comparing(Booster::getName));
```

Comment:
There are no good concurrent implementations for `List`. When you read the docs they point you to the `Set` and `Map` versions and advise you to try and use those instead of lists.

And as I suggested above, we could return a `Collection` instead of a `List` to avoid the allocation.